hr_period/models/hr_period.py

```diff
@@ -0,0 +1,147 @@
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+
+from .hr_fiscal_year import get_schedules
+
+
+class HrPeriod(models.Model):
+    _name = 'hr.period'
+    _description = 'HR Payroll Period'
+    _order = 'date_start'
+
+    name = fields.Char(
+        'Name',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    number = fields.Integer(
+        'Number',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    date_start = fields.Date(
+        'Start of Period',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    date_stop = fields.Date(
+        'End of Period',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    date_payment = fields.Date(
+        'Date of Payment',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    fiscalyear_id = fields.Many2one(
+        'hr.fiscalyear',
+        'Fiscal Year',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        ondelete='cascade'
+    )
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('open', 'Open'),
+            ('done', 'Closed')
+        ],
+        'Status',
+        readonly=True,
+        required=True,
+        default='draft'
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        string='Company',
+        store=True,
+        related="fiscalyear_id.company_id",
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    schedule_pay = fields.Selection(
+        get_schedules,
+        'Scheduled Pay',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    payslip_ids = fields.One2many(
+        'hr.payslip',
+        'hr_period_id',
+        'Payslips',
+        readonly=True
+    )
+
+    @api.model
+    def get_next_period(self, company_id, schedule_pay):
+        """
+         Get the next payroll period to process
+        :rtype: hr.period browse record
+        """
+        periods = self.search([
+            ('company_id', '=', company_id),
+            ('schedule_pay', '=', schedule_pay),
+            ('state', '=', 'open'),
+        ], order='date_start', limit=1)
+
+        return periods[0] if len(periods) else False
```

Comment:
return period

you already pass limit=1 in your query. periods will either contain 1 record or be an empty record set